torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2192,6 +2193,44 @@ int hashValue(Stack& stack) {
   return 0;
 }
 
+int64_t getIValueId(const IValue& val) {
+  // id in practice is only used to compare pointer types, and None.
+  // Python gives no guarantees otherwise.
+  if (val.isPtrType()) {
+    return reinterpret_cast<int64_t>(val.internalToPointer());
+  }
+  if (val.isNone()) {
+    return 0;
+  }
+  // For non-None / pointer types, make sure that they don't compute id(0)
+  if (val.isInt()) {
+    // make 2^62 always 1 so id(int) != 0
+    // use value instead of bits so this works on big/little endian machines
```

Comment:
endianness only affects byte order, `1 << N` is the same regardless of big or little endian.